src/irmin-pack/irmin_pack_layers.ml

```diff
@@ -608,7 +608,6 @@ struct
         ~mem_commit_upper:(mem_commit_next t) t.X.Repo.branch
 
     let skip_with_stats ~skip h =
-      pause () >>= fun () ->
```

Comment:
this is causing memory pressure, but maybe we need this for better fairness?

Reply:
Yes, the reason we added the pauses was to ensure that freeze thread is not "blocking" the main thread. I hope to reproduce this scenario in the bench layers once we figure out the bug that stops the benchmarks after a few cycles. 
If you run the benchmarks just with this line commented out, you can see a speed up. So it would be nice to call freeze as little as possible.

Reply:
I run the benchmark with this pause commented out, and I only plotted the time for commits, to try to show the fairness issue:
![no_pauses_commits](https://user-images.githubusercontent.com/16655454/101547574-be02a500-39aa-11eb-9deb-82b5b6b636d4.png)

I only plotted the time for commits -- it's not the freeze that blocks its the first commit after calling freeze:
```
+1607461030us  application call to freeze completed in 0.000015
+1607461043us  application Commit 8f989dfbd574e3e288d25393f25231bff1f2d6c5 0 in cycle completed in 12.653322; objects created: 558
```
if we zoom in we can see that there are several commits that are also taking longer 
![zoom_in](https://user-images.githubusercontent.com/16655454/101547608-c9ee6700-39aa-11eb-82c2-073c88b0b9fc.png)

Reply:
Wow, that makes a big difference in fairness. What about replacing it with something like `Lwt_unix.yield` for every megbyte written (as suggested by https://github.com/ocsigen/lwt/issues/622#issuecomment-541789628). Maybe we can hook this somewhere in the IO module.

Reply:
So the spikes are due to the extra frequency at which `merge` are performed due to the performance gain. At one point, a `replace` will trigger a blocking merge and will block until the previous merge is complete. E.g. it's not a problem of fairness (or at least we are not tracing the right fairness metrics :p)